test/unit/network_spec.js

```diff
@@ -0,0 +1,164 @@
+/* globals expect, it, describe, waitsFor, combineUrl, PDFNetworkStream */
+
+'use strict';
+
+describe('network', function() {
+  var pdf1 = combineUrl(window.location.href, '../pdfs/tracemonkey.pdf');
+  var pdf1Length = 1016315;
+  var pdf2 = combineUrl(window.location.href, '../pdfs/pdf.pdf');
+  var pdf2Length = 32472771;
+
+  function waitsForPromiseResolved(promise, successCallback) {
+    var resolved = false;
+    promise.then(function(val) {
+        resolved = true;
+        successCallback(val);
+      },
+      function(error) {
+        // Shouldn't get here.
+        expect(error).toEqual('the promise should not have been rejected');
+      });
+    waitsFor(function() {
+      return resolved;
+    }, 20000);
+  }
+
+  it('read without stream and range', function() {
+    var stream = new PDFNetworkStream({
+      source: {
+        url: pdf1,
+        rangeChunkSize: 65536,
+        disableStream: true,
+      },
+      disableRange: true
+    });
+
+    var fullReader = stream.getFullReader();
+
+    var disableStream, disableRange;
+    var promise = fullReader.headersReady.then(function () {
+      disableStream = fullReader.disableStream;
+      disableRange = fullReader.disableRange;
+    });
+
+    var len = 0, count = 0;
+    var read = function () {
+      return fullReader.read().then(function (result) {
+        if (result.done) {
+          return;
+        }
+        count++;
+        len += result.value.byteLength;
+        return read();
+      });
+    };
+
+    var readPromise = read();
+
+    waitsForPromiseResolved(readPromise, function (page) {
+      expect(len).toEqual(pdf1Length);
+      expect(count).toEqual(1);
+      expect(disableStream).toEqual(true);
+      expect(disableRange).toEqual(true);
+    });
+  });
+
+  it('read with streaming', function() {
+    var userAgent = window.navigator.userAgent;
+    var m = /Mozilla\/5.0.*?rv:(\d+).*? Gecko/.exec(userAgent);
+    if (!m || m[1] < 9) {
+      return; // atm test is valid for FF only
```

Comment:
Make this a `TODO` and explain why this is only valid for Firefox. E.g. explain which APIs are missing and link to the relevant bugs.